### PR TITLE
Add a handler to gossip message that checks if we already have the block

### PR DESCRIPTION
**Pull request by agunde406**
Signed-off-by: Andrea Gunderson <agunde@bitwise.io>

Also changes the check in chain controller for if we have a block to check the block cache instead of block store

### CODE REVIEW

File: validator/sawtooth_validator/journal/chain.py [hunk 1]
```diff
@@ -849,7 +849,7 @@ def on_block_validated(self, commit_new_block, result):
     def on_block_received(self, block):
         try:
             with self._lock:
-                if block.header_signature in self._block_store:
+                if block.header_signature in self._block_cache:
```
Comment (aludvik) [inline validator/sawtooth_validator/journal/chain.py]: Should this also use `has_block()`?

File: validator/sawtooth_validator/journal/chain.py [hunk 2]
```diff
@@ -889,6 +889,18 @@ def on_block_received(self, block):
             LOGGER.exception(
                 "Unhandled exception in ChainController.on_block_received()")
 
+    def has_block(self, block_id):
```
Comment (aludvik) [inline validator/sawtooth_validator/journal/chain.py]: Does this need to be inside a lock?

File: validator/sawtooth_validator/server/network_handlers.py [hunk 3]
```diff
@@ -214,7 +222,7 @@ def add(
             completer=completer),
         thread_pool)
 
-    # GOSSIP_MESSAGE 6) Send message to completer
+    # GOSSIP_MESSAGE 8) Send message to completer
```
Comment (aludvik) [inline validator/sawtooth_validator/server/network_handlers.py]: Maybe just delete the numbers?

File: validator/sawtooth_validator/journal/chain.py (hunk 1 above)
Comment (agunde406) [reply validator/sawtooth_validator/journal/chain.py]: Yes, I will update

File: validator/sawtooth_validator/journal/chain.py (hunk 2 above)
Comment (agunde406) [reply validator/sawtooth_validator/journal/chain.py]: Yes i will update

File: validator/sawtooth_validator/server/network_handlers.py (hunk 3 above)
Comment (agunde406) [reply validator/sawtooth_validator/server/network_handlers.py]: Sure

**Pull request merged**